### ISSUE TITLE
Strange error messages are thrown when there are unbound resources

### ISSUE THREAD

**Issue by ymark** [isl/x3ml#119]
When we use resources defined as types or relationships (see below) that are not bound to a specific namespace then some strange messages are thrown. Some indicative unbound resources are the following: 

<pre>
<type>:E22_Man-Made_Object</type>
</pre>

or 
<pre>
<relationship>:P1_is_identified_by</relationship>
</pre>

Furthermore it seems that the error messages are different if the info block exists.

If the info block is missing then the error message is: 

<pre>
ERROR Utils:122 - The namespace with prefix "" has not been declared
</pre>

If the info block exists then the error message is: 

<pre>
Exception in thread "main" com.hp.hpl.jena.shared.BadURIException: Only well-formed absolute URIrefs can be included in RDF/XML output: <E22> Code: 57/REQUIRED_COMPONENT_MISSING in SCHEME: A component that is required by the scheme is missing.
	at com.hp.hpl.jena.xmloutput.impl.BaseXMLWriter.checkURI(BaseXMLWriter.java:836)
	at com.hp.hpl.jena.xmloutput.impl.BaseXMLWriter.xmlnsDecl(BaseXMLWriter.java:351)
	at com.hp.hpl.jena.xmloutput.impl.Unparser.printNameSpaceDefn(Unparser.java:1075)
	at com.hp.hpl.jena.xmloutput.impl.Unparser.wRDF(Unparser.java:338)
	at com.hp.hpl.jena.xmloutput.impl.Unparser.write(Unparser.java:247)
	at com.hp.hpl.jena.xmloutput.impl.Abbreviated.writeBody(Abbreviated.java:142)
	at com.hp.hpl.jena.xmloutput.impl.BaseXMLWriter.writeXMLBody(BaseXMLWriter.java:503)
	at com.hp.hpl.jena.xmloutput.impl.BaseXMLWriter.write(BaseXMLWriter.java:475)
	at com.hp.hpl.jena.xmloutput.impl.Abbreviated.write(Abbreviated.java:127)
	at com.hp.hpl.jena.xmloutput.impl.BaseXMLWriter.write(BaseXMLWriter.java:462)
	at com.hp.hpl.jena.rdf.model.impl.ModelCom.write(ModelCom.java:345)
	at eu.delving.x3ml.engine.ModelOutput.writeXML(ModelOutput.java:125)
	at gr.forth.ics.isl.example.Example.main(Example.java:72)
</pre>

The X3ML mappings file for reproducing the issue are attached.
[mappings.x3ml.txt](https://github.com/isl/x3ml/files/1927151/mappings.x3ml.txt)

**Comment by ymark** [comment]
Added extra checks that validate the prefix and the URI of a namespace when they are added in the XPathContext (the structure that stores the namespaces and the actual URIs). 

The validation is carried out in terms of empty values for the prefix and the URI. To this end the following **ERROR** messages are shown: 

When the URI of the namespace is empty: 

```
Invalid namespace declaration: the URI of a namespace cannot be empty [Prefix: "err", URI: (empty)]
```

When the prefix of the namespace is empty: 

```
Invalid namespace declaration: the prefix of a namespace cannot be empty [Prefix: (empty), URI: "http://localhost/"]
```

When both the prefix and the URI of the namespace are empty: 

```
Invalid namespace declaration: the prefix of a namespace cannot be empty [Prefix: (empty), URI: (empty)]
```

In addition, the execution of the transformation **stops**, because otherwise the error will be thrown from Jena (i.e. for non well-formed URIs).

The fix will appear in version 1.8.5

**Comment by ymark** [comment]
The issue has been resolved by validating the values of _types_ and _relationships_. The following comparisons are now supported: 

* if the value is a URI (e.g. http://blabla...) then the type is a fully qualified URI and accept it as it is.
* if the value has a prefix (e.g. crm:E22_Man-Made_Object) then the abbreviation of the namespace is given
* if the namespace abbreviation is missing (e.g. :E22_Man-Made_Object) or the value of the type is unqualified (e.g. E22_Man-Made_Object)then throw an error. In both cases the error is the following: 

`Unqualified tag: 'E41_Appellation'. The namespace of the resource is missing`

The updated error messages hanldling will appear from version 1.8.5
